content/scripts/build.js

```diff
@@ -356,34 +356,34 @@ class Builder {
     return source;
   }
 
-  reportMacroRenderingFlaws() {
-    // The "this.macroRenderer" instance (used within "this.renderMacros")
-    // accumulates the flaws it encounters within its "flaws" property,
-    // a Map of flaws by URI.
-    if (this.macroRenderer.flaws.size) {
-      if (this.options.flaws === FLAWS_LEVELS.IGNORE) {
-        this.logger.info(
-          chalk.yellow.bold(
-            `\nIgnored flaws within ${this.macroRenderer.flaws.size} document(s) while rendering macros.`
-          )
-        );
-      } else {
-        this.logger.error(
-          chalk.red.bold(
-            `\nFlaws within ${this.macroRenderer.flaws.size} document(s) while rendering macros:`
-          )
-        );
-        for (const [uri, flaws] of this.macroRenderer.flaws) {
-          this.logger.error(
-            chalk.red.bold(`*** flaw(s) while rendering ${uri}:`)
-          );
-          for (const flaw of flaws) {
-            this.logger.error(chalk.red(`${flaw}\n`));
+  recordFlaws(type, uri, flaws) {
+    if (!this.flawsByType.has(type)) {
+      this.flawsByType.set(type, new Map());
+    }
+    const flawsByUri = this.flawsByType.get(type);
+    flawsByUri.set(uri, flaws);
+  }
+
+  reportFlaws(type) {
```

Comment:
These are just temporary and meant to be replaced with the flaw-reporting system.